internal/util/util.go

```diff
@@ -68,57 +68,53 @@ var (
 
 // Config holds the parameters list which can be configured.
 type Config struct {
-	Vtype           string // driver type [rbd|cephfs|liveness|controller]
-	Endpoint        string // CSI endpoint
-	DriverName      string // name of the driver
-	DriverNamespace string // namespace in which driver is deployed
-	NodeID          string // node id
-	InstanceID      string // unique ID distinguishing this instance of Ceph CSI
-	PluginPath      string // location of cephcsi plugin
-	StagingPath     string // location of cephcsi staging path
-	DomainLabels    string // list of domain labels to read from the node
-
-	// metrics related flags
-	MetricsPath     string // path of prometheus endpoint where metrics will be available
-	HistogramOption string // Histogram option for grpc metrics, should be comma separated value,
-	// ex:= "0.5,2,6" where start=0.5 factor=2, count=6
-	MetricsIP         string        // TCP port for liveness/ metrics requests
-	PidLimit          int           // PID limit to configure through cgroups")
-	MetricsPort       int           // TCP port for liveness/grpc metrics requests
-	PollTime          time.Duration // time interval in seconds between each poll
-	PoolTimeout       time.Duration // probe timeout in seconds
-	EnableGRPCMetrics bool          // option to enable grpc metrics
-
+	EnableGRPCMetrics  bool // option to enable grpc metrics
 	EnableProfiling    bool // flag to enable profiling
 	IsControllerServer bool // if set to true start provisoner server
 	IsNodeServer       bool // if set to true start node server
 	Version            bool // cephcsi version
-
 	// SkipForceFlatten is set to false if the kernel supports mounting of
 	// rbd image or the image chain has the deep-flatten feature.
 	SkipForceFlatten bool
-
 	// cephfs related flags
 	ForceKernelCephFS bool // force to use the ceph kernel client even if the kernel is < 4.17
+	UpdateMetadata    bool // update PV/PVC/PVCNamespace details on the backend image as metadata
 
 	// RbdHardMaxCloneDepth is the hard limit for maximum number of nested volume clones that are taken before a flatten
 	// occurs
 	RbdHardMaxCloneDepth uint
-
 	// RbdSoftMaxCloneDepth is the soft limit for maximum number of nested volume clones that are taken before a flatten
 	// occurs
 	RbdSoftMaxCloneDepth uint
-
 	// MaxSnapshotsOnImage represents the maximum number of snapshots allowed
 	// on rbd image without flattening, once the limit is reached cephcsi will
 	// start flattening the older rbd images to allow more snapshots
 	MaxSnapshotsOnImage uint
-
```

Comment:
not part of this change can this be reverted back?

Reply:
I did same changes, it's not failing for me. :thinking: 

```go
[🎩︎]mrajanna@fedora ceph-csi $]make go-lint
cephcsi image settings: quay.io/cephcsi/cephcsi version canary
./scripts/lint-go.sh
INFO [config_reader] Used config file scripts/golangci.yml 
INFO [lintersdb] Active 63 linters: [asciicheck bidichk bodyclose cyclop deadcode depguard dogsled durationcheck errcheck errname errorlint exhaustive exportloopref forcetypeassert gocognit goconst gocritic gocyclo godot gofmt gofumpt goheader goimports golint gomodguard goprintffuncname gosec gosimple govet ifshort importas ineffassign interfacer lll makezero maligned misspell nakedret nestif nilerr nlreturn noctx nolintlint paralleltest predeclared promlinter revive rowserrcheck scopelint sqlclosecheck staticcheck structcheck stylecheck tenv thelper tparallel typecheck unconvert unparam unused varcheck wastedassign whitespace] 
INFO [loader] Using build tags: [pacific]         
INFO [loader] Go packages loading at mode 575 (compiled_files|deps|exports_file|imports|name|types_sizes|files) took 3.543044639s 
WARN [runner] The linter 'golint' is deprecated (since v1.41.0) due to: The repository of the linter has been archived by the owner.  Replaced by revive. 
WARN [runner] The linter 'interfacer' is deprecated (since v1.38.0) due to: The repository of the linter has been archived by the owner.  
WARN [runner] The linter 'scopelint' is deprecated (since v1.39.0) due to: The repository of the linter has been deprecated by the owner.  Replaced by exportloopref. 
WARN [runner] The linter 'maligned' is deprecated (since v1.38.0) due to: The repository of the linter has been archived by the owner.  Replaced by govet 'fieldalignment'. 
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 42.586643ms 
INFO [linters context] importas settings found, but no aliases listed. List aliases under alias: key. 
INFO [linters context/goanalysis] analyzers took 1m41.518813007s with top 10 stages: gocritic: 30.226675639s, buildir: 7.597464563s, the_only_name: 7.219152421s, wastedassign: 5.488997292s, buildssa: 5.395790241s, gofumpt: 3.694478959s, goimports: 3.33554207s, gofmt: 2.087923004s, golint: 2.022580381s, gosec: 2.004654201s 
INFO [runner] Issues before processing: 513, after processing: 0 
INFO [runner] Processors filtering stat (out/in): cgo: 513/513, path_prettifier: 513/513, autogenerated_exclude: 513/513, identifier_marker: 513/513, filename_unadjuster: 513/513, skip_dirs: 513/513, exclude: 513/513, exclude-rules: 181/513, nolint: 0/181, skip_files: 513/513 
INFO [runner] processing took 85.550267ms with stages: nolint: 46.458945ms, exclude-rules: 12.957937ms, identifier_marker: 12.678271ms, autogenerated_exclude: 11.027662ms, path_prettifier: 2.031872ms, skip_dirs: 290.513µs, filename_unadjuster: 50.715µs, cgo: 47.344µs, max_same_issues: 2.278µs, skip_files: 801ns, uniq_by_line: 638ns, severity-rules: 614ns, max_from_linter: 487ns, diff: 411ns, source_code: 379ns, max_per_file_from_linter: 365ns, exclude: 335ns, sort_results: 259ns, path_shortener: 241ns, path_prefixer: 200ns 
INFO [runner] linters took 13.373613913s with stages: goanalysis_metalinter: 13.287932562s 
INFO File cache stats: 116 entries of total size 1.0MiB 
INFO Memory: 157 samples, avg is 779.2MB, max is 1382.8MB 
INFO Execution took 16.972425146s                 
[🎩︎]mrajanna@fedora ceph-csi $]git diff
diff --git a/cmd/cephcsi.go b/cmd/cephcsi.go
index f419790b0..d8b200538 100644
--- a/cmd/cephcsi.go
+++ b/cmd/cephcsi.go
@@ -117,6 +117,11 @@ func init() {
                "maxsnapshotsonimage",
                450,
                "Maximum number of snapshots allowed on rbd image without flattening")
+       flag.BoolVar(
+               &conf.SetMetadata,
+               "setmetadata",
+               false,
+               "set metadata on the volume")
        flag.UintVar(
                &conf.MinSnapshotsOnImage,
                "minsnapshotsonimage",
diff --git a/internal/util/util.go b/internal/util/util.go
index c67fd0570..2995be051 100644
--- a/internal/util/util.go
+++ b/internal/util/util.go
@@ -101,6 +101,8 @@ type Config struct {
        // cephfs related flags
        ForceKernelCephFS bool // force to use the ceph kernel client even if the kernel is < 4.17
 
+       SetMetadata bool // set metadata on the volume
+
        // RbdHardMaxCloneDepth is the hard limit for maximum number of nested volume clones that are taken before a flatten
        // occurs
        RbdHardMaxCloneDepth uint
[🎩︎]mrajanna@fedora ceph-csi $]
```